js/components/ChartForm.vue

```diff
@@ -410,12 +446,154 @@ export default {
       this[vModelField] = target.value;
     },
 
+    /**
+     * Validates the form on input and change events. Closely-spaced events are coalesced.
+     */
+    validateChanges: debounce(function(evt) {
+      const { target } = evt;
+      this.validateInput(target);
+    }, 300),
+
+    /**
+     * Validates all the form inputs prior to saving.
+     */
+    validateForm() {
+      const { $el } = this;
+      const inputs = $el.querySelectorAll(selector.validatedInputs);
```

Comment:
According to MDN, [not all browsers support using `forEach` on a `NodeList`](https://developer.mozilla.org/en-US/docs/Web/API/NodeList), hence the `for` loop.